dev-python/spyder/spyder-4.0.1.ebuild

```diff
@@ -0,0 +1,141 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{6,7} )
+
+inherit eutils distutils-r1 virtualx xdg-utils
+
+# Commit of documentation to fetch
+DOCS_PV="6177401"
+
+DESCRIPTION="The Scientific Python Development Environment"
+HOMEPAGE="
+	https://www.spyder-ide.org/
+	https://github.com/spyder-ide/spyder/
+	https://pypi.org/project/spyder/"
+SRC_URI="https://github.com/spyder-ide/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz
+	https://github.com/spyder-ide/${PN}-docs/archive/${DOCS_PV}.tar.gz -> ${P}-docs.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+RDEPEND="
+	dev-python/atomicwrites[${PYTHON_USEDEP}]
+	>=dev-python/chardet-2.0.0[${PYTHON_USEDEP}]
+	dev-python/cloudpickle[${PYTHON_USEDEP}]
+	dev-python/diff-match-patch[${PYTHON_USEDEP}]
+	dev-python/intervaltree[${PYTHON_USEDEP}]
+	~dev-python/jedi-0.14.1[${PYTHON_USEDEP}]
+	dev-python/keyring[${PYTHON_USEDEP}]
+	dev-python/nbconvert[${PYTHON_USEDEP}]
+	dev-python/numpydoc[${PYTHON_USEDEP}]
+	dev-python/pexpect[${PYTHON_USEDEP}]
+	dev-python/pickleshare[${PYTHON_USEDEP}]
+	dev-python/psutil[${PYTHON_USEDEP}]
+	>=dev-python/pygments-2.0[${PYTHON_USEDEP}]
+	dev-python/pylint[${PYTHON_USEDEP}]
+	>=dev-python/python-language-server-0.31.2[${PYTHON_USEDEP}]
+	>=dev-python/pyxdg-0.26[${PYTHON_USEDEP}]
+	dev-python/pyzmq[${PYTHON_USEDEP}]
+	>=dev-python/qdarkstyle-2.7[${PYTHON_USEDEP}]
+	>=dev-python/qtawesome-0.5.7[${PYTHON_USEDEP}]
+	>=dev-python/qtconsole-4.6.0[${PYTHON_USEDEP}]
+	>=dev-python/QtPy-1.5.0[${PYTHON_USEDEP},svg,webengine]
```

Comment:
I got some errors in test_phase that was resolved with `dev-python/PyQt5[webkit]` 

Then I started running into
```
============================================ ERRORS ============================================
___ ERROR collecting spyder/plugins/variableexplorer/widgets/tests/test_collectioneditor.py ____
spyder/plugins/variableexplorer/widgets/tests/test_collectioneditor.py:26: in <module>
    import pandas
/usr/lib64/python3.6/site-packages/pandas/__init__.py:26: in <module>
    from pandas._libs import (hashtable as _hashtable,
/usr/lib64/python3.6/site-packages/pandas/_libs/__init__.py:4: in <module>
    from .tslibs import (
/usr/lib64/python3.6/site-packages/pandas/_libs/tslibs/__init__.py:7: in <module>
    from .period import Period, IncompatibleFrequency
stringsource:105: in init pandas._libs.tslibs.period
    ???
E   AttributeError: type object 'pandas._libs.tslibs.period.array' has no attribute '__reduce_cython__'
____ ERROR collecting spyder/plugins/variableexplorer/widgets/tests/test_dataframeeditor.py ____
spyder/plugins/variableexplorer/widgets/tests/test_dataframeeditor.py:25: in <module>
    from pandas import (DataFrame, date_range, read_csv, concat, Index, RangeIndex,
/usr/lib64/python3.6/site-packages/pandas/__init__.py:26: in <module>
    from pandas._libs import (hashtable as _hashtable,
/usr/lib64/python3.6/site-packages/pandas/_libs/__init__.py:4: in <module>
    from .tslibs import (
/usr/lib64/python3.6/site-packages/pandas/_libs/tslibs/__init__.py:7: in <module>
    from .period import Period, IncompatibleFrequency
stringsource:105: in init pandas._libs.tslibs.period
    ???
E   AttributeError: type object 'pandas._libs.tslibs.period.array' has no attribute '__reduce_cython__'
___ ERROR collecting spyder/plugins/variableexplorer/widgets/tests/test_namespacebrowser.py ____
spyder/plugins/variableexplorer/widgets/tests/test_namespacebrowser.py:22: in <module>
    from spyder.plugins.variableexplorer.widgets.tests.test_collectioneditor import data_table
<frozen importlib._bootstrap>:971: in _find_and_load
    ???
<frozen importlib._bootstrap>:955: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:656: in _load_unlocked
    ???
<frozen importlib._bootstrap>:626: in _load_backward_compatible
    ???
/usr/lib64/python3.6/site-packages/_pytest/assertion/rewrite.py:304: in load_module
    exec(co, mod.__dict__)
spyder/plugins/variableexplorer/widgets/tests/test_collectioneditor.py:26: in <module>
    import pandas
/usr/lib64/python3.6/site-packages/pandas/__init__.py:26: in <module>
    from pandas._libs import (hashtable as _hashtable,
/usr/lib64/python3.6/site-packages/pandas/_libs/__init__.py:4: in <module>
    from .tslibs import (
/usr/lib64/python3.6/site-packages/pandas/_libs/tslibs/__init__.py:7: in <module>
    from .period import Period, IncompatibleFrequency
stringsource:105: in init pandas._libs.tslibs.period
    ???
E   AttributeError: type object 'pandas._libs.tslibs.period.array' has no attribute '__reduce_cython__'
```

And no idea how to fix them. When emerged with `FEATURES="-test"` program seems to work, but might get some runtime errors? Could you verify if `PyQt5[webkit]` is really needed, or is it just some remnant of untouched test files?

Also why not link `/usr/bin/spyder -> /usr/bin/spyder3`?

Reply:
> When emerged with FEATURES="-test" program seems to work, but might get some runtime errors? Could you verify if PyQt5[webkit] is really needed, or is it just some remnant of untouched test files?

It looks like QtPy[webkit]/PyQt5[webkit] is only required by the tests. I just did some work in spyder just now with webkit support removed from QtPy/PyQt5, and it worked just fine.
What to do now though? Should I add `DEPEND="test? ( QtPy[webkit] ) `, because apparently webkit is supposed to get removed from the tree? Or should I just remove the tests that try to import QtWebKit?

Reply:
Latter, if possible. I believe a version bump can be merged with webkit support, but would rather get rid of it sooner than later.

Reply:
I've added some lines to the ebuild to remove the problematic tests. I have tried a patch, however, repoman complained that the file size of my patch was to big, so I gave up on that and just added some lines to the python_prepare_all phase

Reply:
I could host the patch if you wish to, however looks like you already pushed your changes.

However you prefer.

Reply:
> I could host the patch if you wish to, however looks like you already pushed your changes.
>
> However you prefer.

Let's just do it like this for now, there already is quite a lot of lines in python_prepare_all and I'd prefer to keep it mostly in one place. That way I can just comment out the lines in the ebuild one by one when there is a version bump to test if those tests are still broken.